pkg/util/util.go

```diff
@@ -70,3 +72,18 @@ func (r *CountingReader) Read(p []byte) (n int, err error) {
 func (r *CountingReader) Close() error {
 	return r.Reader.Close()
 }
+
+// GetAvailableSpace gets the amount of available space at the path specified.
+func GetAvailableSpace(path string) int64 {
+	var stat syscall.Statfs_t
+	syscall.Statfs(path, &stat)
+	return int64(stat.Bavail) * int64(stat.Bsize)
```

Comment:
It looks like you forgot about filesystem overhead. This is a good place to subtract the 5% overhead.

Reply:
Looks good but let's see a test for this function to make sure.  You could consume some storage space on the FS and ensure that this function returns the precise amount left.

Reply:
Done